packages/zoe/src/zoeService/escrowStorage.js

```diff
@@ -20,9 +20,24 @@ export const makeEscrowStorage = () => {
   const brandToPurse = makeWeakStore('brand');
 
   /** @type {CreatePurse} */
-  const createPurse = (issuer, brand) => {
+  const createPurse = async (issuer, brand) => {
+    if (brandToPurse.has(brand)) {
+      return;
+    }
+    let purseP;
+    try {
+      purseP = await E(issuer).makeEmptyPurse();
+    } catch (err) {
```

Comment:
The `try/await/catch` seems like a very heavy hammer to get the reason from a rejected promise. More straightforward would be something like

```suggestion
  const createPurse = (issuer, brand) => {
    if (brandToPurse.has(brand)) {
      return undefined; // or Promise.resolve(undefined) without CreatePurse reform below
    }
    return E.when(
      E(issuer).makeEmptyPurse(),
      purseP => {
        // Check again after the `await`
        if (!brandToPurse.has(brand)) {
          brandToPurse.init(brand, purseP);
        }
      },
      err =>
        // ...See below for more on the error case...
        assert.fail(
          X`A purse could not be created for brand ${brand} because: ${err}`,
        ),
    );
  };
```

Reply:
This looks much better, thanks!